src/main/java/bio/terra/service/snapshot/flight/create/CreateSnapshotFirecloudGroupStep.java

```diff
@@ -0,0 +1,46 @@
+package bio.terra.service.snapshot.flight.create;
+
+import bio.terra.service.auth.iam.IamService;
+import bio.terra.service.auth.iam.exception.IamConflictException;
+import bio.terra.service.duos.DuosService;
+import bio.terra.service.snapshot.flight.SnapshotWorkingMapKeys;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import java.util.UUID;
+
+public class CreateSnapshotFirecloudGroupStep implements Step {
+  private final IamService iamService;
+  private final UUID snapshotId;
+
+  public CreateSnapshotFirecloudGroupStep(IamService iamService, UUID snapshotId) {
+    this.iamService = iamService;
+    this.snapshotId = snapshotId;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+
+    String groupName = DuosService.constructFirecloudGroupName(String.valueOf(snapshotId));
+    try {
+      iamService.createGroup(groupName);
+    } catch (IamConflictException ex) {
+      groupName = DuosService.constructUniqueFirecloudGroupName(String.valueOf(snapshotId));
+      iamService.createGroup(groupName);
+    }
```

Comment:
this is duplicated logic from DuosService.createFirecloudGroup. I didn't want the groupEmail or the DuosFirecloudGroupModel. Maybe theres a better way to share more of the code.

Reply:
I could extend the use of the DuosFirecloudGroupModel and rename it but that seemed like a lot

Reply:
I decided to make a new object that retruns the groupName and groupEmail and share the code